lib/rules/selector-class-pattern/index.mjs

```diff
@@ -123,32 +123,31 @@ const rule = (primary, secondaryOptions) => {
  */
 function hasInterpolatingAmpersand(selector) {
 	for (const [i, char] of Array.from(selector).entries()) {
-		if (char !== '&') {
-			continue;
-		}
-
-		const prevChar = selector.charAt(i - 1);
-
-		if (prevChar && !isCombinator(prevChar)) {
-			return true;
-		}
+		if (char !== '&') continue;
 
-		const nextChar = selector.charAt(i + 1);
-
-		if (nextChar && !isCombinator(nextChar)) {
-			return true;
-		}
+		if (!hasAdjacentCombinator(selector, i)) return true;
 	}
 
 	return false;
 }
 
 /**
- * @param {string} x
- * @returns {boolean}
+ * @param {string} selector
+ * @param {number} index
  */
-function isCombinator(x) {
-	return /[\s+>~]/.test(x);
```

Comment:
Is it not sufficient to add `|` to this regular expression?

Are there any cases where that would be incorrect?

I tested this change locally and it did seem to work fine. It would be nice if we can reduce this entire change to just :


```diff
/**
 * @param {string} x
 * @returns {boolean}
 */
function isCombinator(x) {
-	return /[\s+>~]/.test(x);
+	return /[\s+>~|]/.test(x);
}
```

Reply:
Even when it does clash, how would that be an issue?

`&|foo` or `svg|&` can never form a different class name after resolving the nested selectors.

Only things like `&__foo` can become something like `.bar__foo` after apply sass-like nested selector resolving.

Reply:
I am a bit busy right now but Ill address your review in the coming weeks.